src/input-helper.ts

```diff
@@ -241,7 +241,10 @@ async function getCommitMessagesFromPullRequest(
           edges {
             node {
               commit {
-                message
+                message,
+                parents(last: 1) {
+                  totalCount
+                }
```

Comment:
Here's the start of our change, extending the query in graphQL to choose the total count of parents for the given commit. As noted, merge commits show two parents, so this is our indicator for filtering those out of the check.